src/Microsoft.Tye.Hosting/DockerRunner.cs

```diff
@@ -195,6 +195,20 @@ private async Task StartContainerAsync(Application application, Service service,
             var volumes = "";
             var workingDirectory = docker.WorkingDirectory != null ? $"-w {docker.WorkingDirectory}" : "";
             var hostname = "host.docker.internal";
+            var dockerImage = docker.Image ?? service.Description.Name;
+
+            if (docker.DockerFile != null)
+            {
+                var exitCode = await ProcessUtil.RunAsync(
+                    $"docker",
+                    $"build \"{docker.DockerFile.DirectoryName}\" -t {dockerImage} -f \"{docker.DockerFile}\"",
```

Comment:
We need to support a different build context as well.

Reply:
How would you recommend exposing that? In the tye.yaml as another part of a service?

Reply:
Yep